testng-core/src/main/java/org/testng/TestNG.java

```diff
@@ -1403,7 +1412,12 @@ public static TestNG privateMain(String[] argv, ITestListener listener) {
    *
    * @param cla The command line parameters
    */
-  protected void configure(CommandLineArgs cla) {
+  public void configure(CommandLineArgs cla) {
+    boolean skipTestFlag =
+        Optional.ofNullable(cla.includeAllDataDrivenTestsWhenSkipping).orElse(false);
```

Comment:
Logic duplication: you already set the default value on `CommandLineArgs`. Maybe it should be a `boolean` in `CommandLineArgs` or null by default there.

Reply:
In fact, why not follow the current pattern with null check?

Reply:
I will revisit the false part and check. I added the additional guards because tests were throwing a NullPointerException if we do a null check. 

On a side note do you see any problem with us using 'Optional.ofNullable()'?

Reply:
> On a side note do you see any problem with us using 'Optional.ofNullable()'?

I think `Optional` is good for API design. I don't know yet if it is a good or bad idea to use it for internal.